components/shared/Carousel.js

```diff
@@ -0,0 +1,74 @@
+// @author William Shi
+
+import { Carousel } from 'react-bootstrap'
+import Image from 'next/image'
+
+/**
+ * A carousel, or a slideshow
+ * @param {React.Component} slides - the slides to display
+ */
+
+const AppCarousel = ({ slides }) => {
+  let carousel = []
+  console.log(slides)
+  for (var i = 0; i < slides.length; i = i + 3) {
+    console.log(i)
+    console.log(slides.length)
+    if (slides.length % 3 === 1 && i + 3 >= slides.length) {
+      carousel[i] = (
+        <Carousel.Item key={i}>
+          <div className="row justify-content-center">
+            <img
+              className="d-block w-30"
+              src={slides[i].img}
+              alt={slides[i].alt}
+            />
+          </div>
+        </Carousel.Item>
+      )
+    } else if (slides.length % 3 === 2 && i + 3 >= slides.length) {
```

Comment:
@jimmygit Would the input already need to be split into chunks of 3? I think it would be a bad abstraction if it you need to already split into chunks of 3. The user of this component shouldn't need to think about this.

Reply:
@OrenLeung No the input can stay as 1D. I was talking about having logic that does the splitting.

Reply:
agree with jimmy, it's a lot easier.

Reply:
ohh that makes sense! @jimmyygit